### PR TITLE
Make small lightable objects not work in an environment with little oxygen

### LINKED ISSUE #14417: No oxygen yet the campfire burns
(WEB REPORT BY: mchotnspicy REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)

26565f98f4ebbae6551ecb651f1ca058f0724828 
>General description of the issue
no pressure but the campfire burns just fine

>What you expected to happen
campfire dies out

>What actually happened
burn baby :^)))

>Steps to reproduce if possible
break the wall then light the fire

### PR DESCRIPTION
**Pull request by ghost**
Fixes #14417

Makes sense, right? 
No, I didn't touch welders.

:cl:
* tweak: You can no longer light lighters, matches, campfires, or cigarettes in space.

### CODE REVIEW

**Comment by BobdaBiscuit** [comment]
>space heater needing oxygen to heat stuff up

Is it even fire based or is it something like a quartz/element based?

**Comment by ghost** [comment]
OH SHIT i meant to do that to the campfire type not the space heater lemme fix

fixed

**Comment by unusualcrow** [comment]
aw fuck build failed

File: code/ATMOSPHERICS/hvac/spaceheater.dm [hunk 1]
```diff
@@ -125,7 +125,9 @@
 
 /obj/machinery/space_heater/campfire/attackby(obj/item/I, mob/user)
 	..()
-	if(!on && cell.charge > 0)
+	var/turf/T = get_turf(src)
+	var/datum/gas_mixture/env = T.return_air()
+	if(!on && cell.charge > 0 && env.oxygen >= 5)
```
Comment (Probe1) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: 1.  Why is a campfire checking for cell charge what is this lazy copy paste shit.

2.  There should be feedback to the player that it didn't work and why.

**Review by Probe1 — CHANGES_REQUESTED** [review]
Add player feedback about the flame failing in some In Character language.

File: code/ATMOSPHERICS/hvac/spaceheater.dm (hunk 1 above)
Comment (PJB3005) [reply code/ATMOSPHERICS/hvac/spaceheater.dm]: The cell charge is because of celt. That's just how the campfire works.

**Comment by CptWad** [comment]
**The flame on the [src] goes out in a weak fashion.**

**Comment by IAGFPRMF** [comment]
Sounds cool.

File: code/ATMOSPHERICS/hvac/spaceheater.dm (hunk 1 above)
Comment (Probe1) [reply code/ATMOSPHERICS/hvac/spaceheater.dm]: Yeah. I'm still not gonna miss an opportunity to point out _what the fuck is this_.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -85,13 +80,19 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
 	var/datum/gas_mixture/env = location.return_air()
-	if(smoketime <= 0 | env.oxygen < 5)
+	if(smoketime <= 0)
 		lit = -1
 		update_brightness()
 		return
+	if(env.oxygen < 5)
+		lit = -1
+		update_brightness()
+		to_chat(M, "The flame on the [src] suddenly goes out in a weak fashion.")
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm]: `\the`

File: code/ATMOSPHERICS/hvac/spaceheater.dm [hunk 3]
```diff
@@ -125,6 +125,11 @@
 
 /obj/machinery/space_heater/campfire/attackby(obj/item/I, mob/user)
 	..()
+	var/turf/T = get_turf(src)
+	var/datum/gas_mixture/env = T.return_air()\
```
Comment (PJB3005) [inline code/ATMOSPHERICS/hvac/spaceheater.dm]: Why is there a backslash here?

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 2 above)
Comment (PJB3005) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Also you're not guaranteed to have a holder, are you?

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -341,14 +339,18 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+	if(smoketime <= 0 | env.oxygen < 5)
 		if(!inside_item)
 			var/atom/new_butt = new type_butt(location) //Spawn the cigarette butt
 			transfer_fingerprints_to(new_butt)
 		lit = 0 //Actually unlight the cigarette so that the lighting can update correctly
 		update_brightness()
 		if(ismob(loc))
-			to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
+			if(env.oxygen < 5)
+				to_chat(M, "<span class='notice'>The [name] suddenly goes out in a weak fashion.</span>")
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm]: `\The [src]`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/ATMOSPHERICS/hvac/spaceheater.dm (hunk 3 above)
Comment (ghost) [reply code/ATMOSPHERICS/hvac/spaceheater.dm]: typo

**Review by PJB3005 — APPROVED** [review]

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 5]
```diff
@@ -341,14 +339,18 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+	if(smoketime <= 0 | env.oxygen < 5)
 		if(!inside_item)
 			var/atom/new_butt = new type_butt(location) //Spawn the cigarette butt
 			transfer_fingerprints_to(new_butt)
 		lit = 0 //Actually unlight the cigarette so that the lighting can update correctly
 		update_brightness()
 		if(ismob(loc))
-			to_chat(M, "<span class='notice'>Your [name] goes out.</span>")
+			if(env.oxygen < 5)
+				to_chat(M, "<span class='notice'>\The [name] suddenly goes out in a weak fashion.</span>")
```
Comment (Exxion) [inline code/game/objects/items/weapons/cigs_lighters.dm]: `[src]`, not `[name]`
But only for this one

**Review by Probe1 — APPROVED** [review]

**Comment by Zennyization** [comment]
What about candles buddy?

**Comment by ghost** [comment]
>What about candles buddy?

done

**Pull request merged**